joomla/joomla-cms
# [4.0] Disable lazyloading on images that have no dimensions or added on the first page view

Pull Request for Issue #29250

### Summary of Changes

Disable lazyloading on images that have no dimensions or get added on the first page view

- intro images and full images should be at 99% of the time one of the first things we have on that page and that we don't have dimensions on that we should not try to lazyload them.
- the opt-out stuff via the plugin and the HTMLHelper methods has been switched to only add that tag when height and width attributes are present.

### Testing Instructions

- add inline images to an article that dont have dimensions setup in the image tag
- add inline images with the dimensions setup correctly
- check the frontend and notice that they are lazyloaded (meaning the lazyload tag is added there)
- apply this patch
- notice that this is no longer the case

### Actual result BEFORE applying this Pull Request

- all images also without height and width attributes has been lazyloaded 

### Expected result AFTER applying this Pull Request

- per default only images with height and width attributes are now lazyloaded

### Documentation Changes Required

none

cc @dgrammatiko given that you wanted to work on that too.
